pkg/authn/identityheaders/identityheaders.go

```diff
@@ -52,7 +53,31 @@ func WithAuthHeaders(handler http.Handler, cfg *AuthnHeaderConfig) http.Handler
 			// Seemingly well-known headers to tell the upstream about user's identity
 			// so that the upstream can achieve the original goal of delegating RBAC authn/authz to kube-rbac-proxy
 			req.Header.Set(cfg.UserFieldName, u.GetName())
-			req.Header.Set(cfg.GroupsFieldName, strings.Join(u.GetGroups(), cfg.GroupSeparator))
+
+			if cfg.GroupSeparator == "" {
+				// Add each group as a separate header entry
+				for _, group := range u.GetGroups() {
+					if strings.Contains(group, ",") {
```

Comment:
How does Golang handle a comma in headers?

Reply:
https://goplay.tools/snippet/n-xUX0e0w5K

I think we might be ok leaving even groups with `,` in

Reply:
Thanks a lot for digging into this!

From what I’ve found, go doesnt treat commas inside a single header as a special separator -- it just stores and sends the value as is and combines multiple values into a single header line if needed. Also there is no documentation stating that golang removes or parses commas in any special way, at least I couldn't find it.. 

So since your code shows that the header round-trips correctly -- I agree that we don’t need to drop groups that contain commas. It should be safe to leave them in.